### PR TITLE
Top and middle banners for HoC 2017

### PR DESCRIPTION
**Pull request by chrisgarrity**
Css animated top banner

Middle banner of activities - responsive

Also updated ttt see-more/open-modal link to be consistent with the new icon from Carl.

fixes https://github.com/LLK/scratchr2/issues/4869

Testing:
- [ ] Page has top and middle banners as shown in the issue
- [ ] Top banner
  - [ ] 'Scratch' in the top banner is animated
  - [ ] Animation in the top banner goes to 'animate a name' tutorial when clicked
  - [ ] Top banner button says 'Animate a Name' when logged in and 'Get Started with Coding!' when not logged in
  - [ ] 'See cards and guide' link opens 'Animate a Name' modal
  - [ ] Top banner is responsive and shrinks appropriately for smaller screen sizes
- [ ] Middle banner 
  - [ ] Shows three activities in desktop mode (Make it Fly, Make Music, Pong Game)
  - [ ] Shows two activities in tablet mode  (Make it Fly, Make Music)
  - [ ] is hidden when smaller than tablet mode (i.e. if not big enough to show two activities)
  - [ ] Clicking on the bottom section of an activity opens the cards/guides modal
  - [ ] Clicking anywhere else on an activity opens the editor with the corresponding tutorial open.

### CODE REVIEW

File: src/views/splash/hoc/middle-banner.jsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                bannerUrl: tiles[7].bannerUrl
+            }
+        };
+        
+        return [
+            <TTTTile
+                key={1}
+                className="mod-banner"
+                onGuideClick={this.showTTTModal.bind(this, tileObjects.flyTile)}
+                {...tileObjects.flyTile}
```
Comment (mewtaylor) [inline src/views/splash/hoc/middle-banner.jsx]: I didn't realize this works outside of es6! cool!

File: src/views/splash/hoc/top-banner.jsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        this.setState({TTTModalOpen: false});
+    },
+    render: function () {
+        return (
+            <TitleBanner className="mod-splash-top">
+                <FlexRow className="banner-top inner">
+                    <FlexRow className="top-animation">
+                        <img
+                            src="/images/hoc/s.png"
+                            alt="C"
```
Comment (mewtaylor) [inline src/views/splash/hoc/top-banner.jsx]: I think we can give null `alt`s for these, if we're using them mostly for aria standards: https://www.w3.org/WAI/tutorials/images/decorative/

File: src/views/splash/hoc/top-banner.jsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                        <img
+                            src="/images/hoc/c2.png"
+                            alt="C"
+                            className="top-animation-letter mod-letter-c2"
+                        />
+                        <img
+                            src="/images/hoc/h.png"
+                            alt="H"
+                            className="top-animation-letter mod-letter-h"
+                        />
```
Comment (mewtaylor) [inline src/views/splash/hoc/top-banner.jsx]: @carljbowman @chrisgarrity I think the letters are showing up a little bit pixel-y around the edges when I view on a browser:
screen shot 2017-11-07 at 4 59 08 pm

File: src/components/ttt-tile/ttt-tile.scss [hunk 4]
```diff
@@ -105,14 +105,15 @@
...
     }
 }
 
-.ttt-tile-see-more {
+.ttt-tile-open-modal {
     display: inline-block;
     padding: 0 .25rem;
-    vertical-align: middle;
+    width: 1.25em;
+    height: 1.25em;
```
[the remark this answers is not in the document]
Comment (chrisgarrity) [reply src/components/ttt-tile/ttt-tile.scss]: 👍

File: src/views/splash/hoc/top-banner.jsx (hunk 3 above)
Comment (chrisgarrity) [reply src/views/splash/hoc/top-banner.jsx]: @carljbowman do you want to update the letter images?

**Comment by chrisgarrity** [comment]
@ntlrsk @ChampikaF can you check the copy on the page for the top and middle HoC banners. 

We changed the wording on the top-banner button (Animate a Name so that it's consistent with the link target instead of 'Try to animate a word')

screencapture-localhost-8333-1510149711976

File: src/views/splash/hoc/top-banner.jsx (hunk 3 above)
Comment (chrisgarrity) [reply src/views/splash/hoc/top-banner.jsx]: From @carljbowman The letters are intended to have textured edges, and the texture might just be right on the edge of appearing pixelated and texture on smaller screens. We'll leave them as is for now, but keep it in mind for the future.

Comment (mewtaylor) [reply src/views/splash/hoc/top-banner.jsx]: @chrisgarrity @carljbowman awesome thanks!

**Comment by ChampikaF** [comment]
LGTM!

**Comment by VutonDesign** [comment]
Wow, That looks great!

**Comment by GreenBayRules** [comment]
Whoa! That looks amazing, and probably even cooler animated! Congrats :D

**Comment by chrisgarrity** [comment]
@mewtaylor I've made the new changes to change the  call-to-action based on logged-in state. Could you re-check - new version has been merged to develop.

**Comment by GreenBayRules** [comment]
When will this be added?

**Comment by rschamp** [comment]
QA: Exercise above test cases on these environments

Windows 10
* Chrome (@fsih)
* IE 11 (@cwillisf)
* Edge (@cwillisf)
* FF (@fsih)

ChromeOS
* Chrome (@thisandagain)

MacOS
* Chrome (@kchadha)
* Safari (@rschamp)
* Firefox (@ericrosenbaum)

**Comment by fsih** [comment]
Windows 10 Chrome Version 62.0.3202.94 (Official Build) (64-bit)
Testing:
- [ ] Page has top and middle banners as shown in the issue
- [ ] Top banner
  - [ ] 'Scratch' in the top banner is animated
  - [ ] Animation in the top banner goes to 'animate a name' tutorial when clicked
  - [ ] Top banner button says 'Animate a Name' when logged in and 'Get Started with Coding!' when not logged in
  - [ ] 'See cards and guide' link opens 'Animate a Name' modal
  - [ ] Top banner is responsive and shrinks appropriately for smaller screen sizes
- [ ] Middle banner 
  - [ ] Shows three activities in desktop mode (Make it Fly, Make Music, Pong Game)
  - [ ] Shows two activities in tablet mode  (Make it Fly, Make Music)
  - [ ] is hidden when smaller than tablet mode (i.e. if not big enough to show two activities)
  - [ ] Clicking on the bottom section of an activity opens the cards/guides modal
  - [ ] Clicking anywhere else on an activity opens the editor with the corresponding tutorial open.

**Comment by ericrosenbaum** [comment]
Testing Firefox Quantum 57.0 (64-bit):

- [ ] Page has top and middle banners as shown in the issue
- [ ] Top banner
  - [ ] 'Scratch' in the top banner is animated
  - [ ] Animation in the top banner goes to 'animate a name' tutorial when clicked
  - [ ] Top banner button says 'Animate a Name' when logged in and 'Get Started with Coding!' when not logged in
  - [ ] 'See cards and guide' link opens 'Animate a Name' modal
  - [ ] Top banner is responsive and shrinks appropriately for smaller screen sizes
- [ ] Middle banner 
  - [ ] Shows three activities in desktop mode (Make it Fly, Make Music, Pong Game)
  - [ ] Shows two activities in tablet mode  (Make it Fly, Make Music)
  - [ ] is hidden when smaller than tablet mode (i.e. if not big enough to show two activities)
  - [ ] Clicking on the bottom section of an activity opens the cards/guides modal
  - [ ] Clicking anywhere else on an activity opens the editor with the corresponding tutorial open.

**Comment by kchadha** [comment]
Testing:
- [ ] Page has top and middle banners as shown in the issue
- [ ] Top banner
  - [ ] 'Scratch' in the top banner is animated
  - [ ] Animation in the top banner goes to 'animate a name' tutorial when clicked
  - [ ] Top banner button says 'Animate a Name' when logged in and 'Get Started with Coding!' when not logged in
  - [ ] 'See cards and guide' link opens 'Animate a Name' modal
  - [ ] Top banner is responsive and shrinks appropriately for smaller screen sizes
- [ ] Middle banner 
  - [ ] Shows three activities in desktop mode (Make it Fly, Make Music, Pong Game)
  - [ ] Shows two activities in tablet mode  (Make it Fly, Make Music)
  - [ ] is hidden when smaller than tablet mode (i.e. if not big enough to show two activities)
  - [ ] Clicking on the bottom section of an activity opens the cards/guides modal
  - [ ] Clicking anywhere else on an activity opens the editor with the corresponding tutorial open.

**Comment by fsih** [comment]
Windows 10 Firefox 57.0 (64-bit)
Testing:
- [ ] Page has top and middle banners as shown in the issue
- [ ] Top banner
  - [ ] 'Scratch' in the top banner is animated
  - [ ] Animation in the top banner goes to 'animate a name' tutorial when clicked
  - [ ] Top banner button says 'Animate a Name' when logged in and 'Get Started with Coding!' when not logged in
  - [ ] 'See cards and guide' link opens 'Animate a Name' modal
  - [ ] Top banner is responsive and shrinks appropriately for smaller screen sizes
- [ ] Middle banner 
  - [ ] Shows three activities in desktop mode (Make it Fly, Make Music, Pong Game)
  - [ ] Shows two activities in tablet mode  (Make it Fly, Make Music)
  - [ ] is hidden when smaller than tablet mode (i.e. if not big enough to show two activities)
  - [ ] Clicking on the bottom section of an activity opens the cards/guides modal
  - [ ] Clicking anywhere else on an activity opens the editor with the corresponding tutorial open.

**Comment by fsih** [comment]
On both Firefox and Chrome on Windows, the S, T and H move relatively farther when the screen is narrow, to the point that T can overlap the letters next to it:
image
It's not broken, those 3 letters just stand out a lot more

**Comment by fsih** [comment]
In both Firefox and Chrome on windows, cards are very slightly wider than the screen before they disappear:
image

**Comment by fsih** [comment]
The editor is just not loading for me at all in Windows Firefox:
image
Nothing in the console.

**Comment by kchadha** [comment]
On MacOS chrome, when the window is at it's shortest (in terms of height), the scratch banner gets cut off vertically until you make the window narrower (in terms of width). E.g.

image

vs. 

image

vs.

image

**Comment by rschamp** [comment]
MacOS 10.12 Safari 11.0.1
- [X] Page has top and middle banners as shown in the issue
- [X] Top banner
  - [X] 'Scratch' in the top banner is animated
  - [X] Animation in the top banner goes to 'animate a name' tutorial when clicked
  - [X] Top banner button says 'Animate a Name' when logged in and 'Get Started with Coding!' when not logged in
  - [X] 'See cards and guide' link opens 'Animate a Name' modal
  - [X] Top banner is responsive and shrinks appropriately for smaller screen sizes
- [X] Middle banner 
  - [X] Shows three activities in desktop mode (Make it Fly, Make Music, Pong Game)
  - [X] Shows two activities in tablet mode  (Make it Fly, Make Music)
  - [X] is hidden when smaller than tablet mode (i.e. if not big enough to show two activities)
  - [X] Clicking on the bottom section of an activity opens the cards/guides modal
  - [X] Clicking anywhere else on an activity opens the editor with the corresponding tutorial open.

**Comment by chrisgarrity** [comment]
@GreenBayRules it'll get released just before CS Ed Week (Dec 4-8)

**Comment by thisandagain** [comment]
### ChromeOS
- [ ] Page has top and middle banners as shown in the issue
- [ ] Top banner
  - [ ] 'Scratch' in the top banner is animated
  - [ ] Animation in the top banner goes to 'animate a name' tutorial when clicked
  - [ ] Top banner button says 'Animate a Name' when logged in and 'Get Started with Coding!' when not logged in
  - [ ] 'See cards and guide' link opens 'Animate a Name' modal
  - [ ] Top banner is responsive and shrinks appropriately for smaller screen sizes
- [ ] Middle banner 
  - [ ] Shows three activities in desktop mode (Make it Fly, Make Music, Pong Game)
  - [ ] Shows two activities in tablet mode  (Make it Fly, Make Music)
  - [ ] is hidden when smaller than tablet mode (i.e. if not big enough to show two activities)
  - [ ] Clicking on the bottom section of an activity opens the cards/guides modal
  - [ ] Clicking anywhere else on an activity opens the editor with the corresponding tutorial open.

**Comment by cwillisf** [comment]
Windows 10, Internet Explorer 11.726.15063.0

Note: tips do not open if the browser window is too small.

Testing:
- [x] Page has top and middle banners as shown in the issue
- [x] Top banner
  - [x] 'Scratch' in the top banner is animated
  - [x] Animation in the top banner goes to 'animate a name' tutorial when clicked
  - [x] Top banner button says 'Animate a Name' when logged in and 'Get Started with Coding!' when not logged in
  - [x] 'See cards and guide' link opens 'Animate a Name' modal
  - ⛔️ Top banner is responsive and shrinks appropriately for smaller screen sizes
    - Always large (max size?)
- [x] Middle banner 
  - [x] Shows three activities in desktop mode (Make it Fly, Make Music, Pong Game)
  - [x] Shows two activities in tablet mode  (Make it Fly, Make Music)
  - [x] is hidden when smaller than tablet mode (i.e. if not big enough to show two activities)
  - [x] Clicking on the bottom section of an activity opens the cards/guides modal
    - ❓ This works, but the icon implies to me that the content will open in a pop-up window or a new tab. In desktop mode you can see that it's an overlay within the page, but in tablet or smaller it looks full-screen.
  - [x] Clicking anywhere else on an activity opens the editor with the corresponding tutorial open.

**Comment by cwillisf** [comment]
Windows 10, Edge 40.15063.674.0

Note: tips do not open if the browser window is too small.

Testing:
- [x] Page has top and middle banners as shown in the issue
- [x] Top banner
  - [x] 'Scratch' in the top banner is animated
  - [x] Animation in the top banner goes to 'animate a name' tutorial when clicked
  - [x] Top banner button says 'Animate a Name' when logged in and 'Get Started with Coding!' when not logged in
  - [x] 'See cards and guide' link opens 'Animate a Name' modal
  - [x] Top banner is responsive and shrinks appropriately for smaller screen sizes
    - ❓ The very first time I loaded the page, the animated letters were all very small despite my browser window being very large. After resizing the window down and back up, the letters resized correctly. I haven't been able to repro this since then, even with a hard refresh.
- [x] Middle banner 
  - [x] Shows three activities in desktop mode (Make it Fly, Make Music, Pong Game)
  - [x] Shows two activities in tablet mode  (Make it Fly, Make Music)
  - [x] is hidden when smaller than tablet mode (i.e. if not big enough to show two activities)
  - [x] Clicking on the bottom section of an activity opens the cards/guides modal
    - ❓ This works, but the icon implies to me that the content will open in a pop-up window or a new tab. In desktop mode you can see that it's an overlay within the page, but in tablet or smaller it looks full-screen.
  - [x] Clicking anywhere else on an activity opens the editor with the corresponding tutorial open.

**Comment by cwillisf** [comment]
Judging by what @fsih saw as well as what I saw when my letters were stuck small, it seems like the travel distance for the "t" might be calculated according to its full size rather than its current size.

**Comment by Kenny2github** [comment]
@chrisgarrity @rschamp @mewtaylor this is way past milestone and seems to have some stuff going on - any updates?

**Comment by chrisgarrity** [comment]
@Kenny2github this one is being held until CS Ed week (starts Dec 4). It's all ready to go.

**Pull request merged**